src/json.hpp.re2c

```diff
@@ -193,51 +192,62 @@ inline bool operator<(const value_t lhs, const value_t rhs) noexcept
+template<typename T>
 using uncvref_t = typename std::remove_cv<typename std::remove_reference<T>::type>::type;
 
 // Taken from http://stackoverflow.com/questions/26936640/how-to-implement-is-enum-class-type-trait
-template <typename T>
+template<typename T>
 using is_unscoped_enum =
     std::integral_constant<bool, std::is_convertible<T, int>::value and
     std::is_enum<T>::value>;
 
-// Implementation of 2 C++17 constructs: conjunction, negation.
-// This is needed to avoid evaluating all the traits in a condition
-//
-// For example: not std::is_same<void, T>::value and has_value_type<T>::value
-// will not compile when T = void (on MSVC at least)
-// Whereas conjunction<negation<std::is_same<void, T>>, has_value_type<T>>::value
-// will stop evaluating if negation<...>::value == false
-//
-// Please note that those constructs must be used with caution, since symbols can
-// become very long quickly (which can slow down compilation and cause MSVC internal compiler errors)
-// Only use it when you have too (see example ahead)
-template <class...> struct conjunction : std::true_type {};
-template <class B1> struct conjunction<B1> : B1 {};
-template <class B1, class... Bn>
-struct conjunction<B1, Bn...>
-: std::conditional<bool(B1::value), conjunction<Bn...>, B1>::type {};
+/*
+Implementation of 2 C++17 constructs: conjunction, negation. This is needed
+to avoid evaluating all the traits in a condition
+
+For example: not std::is_same<void, T>::value and has_value_type<T>::value
+will not compile when T = void (on MSVC at least). Whereas
+conjunction<negation<std::is_same<void, T>>, has_value_type<T>>::value will
+stop evaluating if negation<...>::value == false
+
+Please note that those constructs must be used with caution, since symbols can
+become very long quickly (which can slow down compilation and cause MSVC
+internal compiler errors). Only use it when you have to (see example ahead).
+*/
+template<class...> struct conjunction : std::true_type {};
+template<class B1> struct conjunction<B1> : B1 {};
+template<class B1, class... Bn>
+struct conjunction<B1, Bn...> : std::conditional<bool(B1::value), conjunction<Bn...>, B1>::type {};
 
-template <class B> struct negation : std::integral_constant < bool, !B::value > {};
+template<class B> struct negation : std::integral_constant < bool, !B::value > {};
 
 // dispatch utility (taken from ranges-v3)
-template <unsigned N> struct priority_tag : priority_tag < N - 1 > {};
+template<unsigned N> struct priority_tag : priority_tag < N - 1 > {};
+template<> struct priority_tag<0> {};
+
 
-template <> struct priority_tag<0> {};
+//////////////////
+// constructors //
+//////////////////
 
 // This is an experiment. I need this to move constructors out of basic_json.
-// I'm sure there is a better way, but this might need a big basic_json refactoring
-template <value_t> struct external_constructor;
+// I'm sure there is a better way, but this might need a big basic_json
+// refactoring
```

Comment:
I think we can remove a part of this comment, we could only keep the part about the 'there might be a better way'